src/azure-cli-testsdk/azure/cli/testsdk/utilities.py

```
@@ -198,3 +199,18 @@ def process_request(self, request):
                     request.body = body.replace(old, new)
 
         return request
+
+
+def retry(func, timeout, interval=10):
+    """
+    Retry func until success.
+    :param timeout: in seconds
+    :param func: function
+    :param interval: polling interval
+    :return:
+    """
+    for i in range(0, timeout, interval):
```

When `timeout` is reached, the exception is swallowed and can't be thrown out.